Stop ignoring merge hash behaviour in inventory (#58460)
ansible/ansible: docs/docsite/rst/porting_guides/porting_guide_2.9.rst

Playbook
========

>  * hash_behaviour now also affects inventory sources, so if you have it set to 'merge' the data you get from inventory might change and you will have to update playbooks accordingly. Those using the default (overwrite) will see no changes as inventory was ignoring this setting.

Reviewer: ```suggestion
 * ``hash_behaviour`` now affects inventory sources. If you have it set to ``merge``, the data you get from inventory might change and you will have to update playbooks accordingly. If you're using the default setting (``overwrite``), you will see no changes. Inventory was ignoring this setting.
```

>  * ``hash_behaviour`` now affects inventory sources. If you have it set to ``merge``, the data you get from inventory might change and you will have to update playbooks accordingly. If you're using the default setting (``overwrite``), you will see no changes. Inventory was ignoring this setting.

Reviewer 2: Looks good. Should this be an inventory specific setting to avoid breaking things?

Author: Unsure this is 'long standing bug' IMHO, but I'm willing to entertain a toggle if needed, I doubt this affects many people (but those are famous last words).
